### ISSUE TITLE
This script doesn't create unique identifier for each emoji

### ISSUE THREAD

**Issue by real19** [amtypaldos/rocketchat-emoji-upload#1]
This script doesn't create unique identifier for each emoji hence the emojis don't get added to Realm since they don't really have a primary key.

Would appreciate if you could add it to your script file

**Comment by real19** [comment]
It looks like some of your custom emojis are returning with the _id property as an object. Checked and got "_id": {"_str":"5bc429276e3a9221cee9fb03"} when it should have been "_id": "5bc429276e3a9221cee9fb03".

**Comment by amtypaldos** [comment]
@real19 I have not worked on this in in a bit, but I believe that the primary key is created on the MongoDB side.  I did not specify a key as I want the DB to create one so there is not a chance of overlap.

**Comment by amtypaldos** [comment]
@real19 With regards to the id_property can you give a more info?

**Comment by real19** [comment]
Yeah , it seems for some reason that when you use the script to upload emojis , mongo creates the ID as an object for the custom emoji , not sure why. It works fine if I upload an emoji manually.  

Could you perhaps write a script we can run to fix that ?  

I have uploaded a lot of emojis. Maybe generate a UUID. 

Thanks.

**Comment by amtypaldos** [comment]
What version of MongoDB and Rocket.chat are you running?

I can try and replicate.

**Comment by real19** [comment]
The Rocket Chat version is 0.68.3 and the MongoDB server version: 4.0.0.

**Comment by real19** [comment]
So I believe the problem was rocket chat android and iOS apps don't like default ID objects created automatically by Mongo unless they are converted to string . 
I ran the following one mongo to create new IDs and converted them to string and it all worked perfectly: 

```
db.rocketchat_custom_emoji.find().forEach(function(emoji) { 
       print("Processing: " + emoji._id);
       var oldEmojiId = emoji._id;
       emoji._id = ObjectId().toString(); 
       db.rocketchat_custom_emoji.insert(emoji);
       db.rocketchat_custom_emoji.remove({_id: oldEmojiId});
});
```

**Comment by amtypaldos** [comment]
Any idea how to ensure that field is a string when it is created?

**Comment by real19** [comment]
When you say, ObjectId().toString() it puts in a string there . So basically , it's best to manually generate the _id using ObjectId().toString() . There shouldn't be an overlap I think. Otherwise, by default mongo creates ObjectId() which is not a string but an Object. And that makes the emoji sad . :D
